mite/cli/test.py

```diff
@@ -83,17 +85,22 @@ def test_scenarios(test_name, opts, scenarios, config_manager):
     controller = Controller(test_name, scenario_manager, config_manager)
     transport = DirectRunnerTransport(controller)
     receiver = DirectReciever()
+    debug_message_output = DebugMessageOutput()
+    receiver.add_listener(debug_message_output.process_message)
     _setup_msg_processors(receiver, opts)
     http_stats_output = _get_http_stats_output(receiver)
     loop = asyncio.get_event_loop()
 
+    has_error = False
+
     async def controller_report():
+        nonlocal has_error
```

Comment:
Rather than this `nonlocal` business (which I find a bit confusing, although in the end I guess it's pretty simple), I think we could do things slightly differently...

1. make `controller_report` return its status directly
2. use `asyncio.create_task` to wrap it in a task (which we save a reference to in a variable), rather than passing `controller_report()` directly to `asyncio.gather`
3. use `task_var.result()` to get the result back

I'm not 100% sure this works (i.e. I haven't tested), but I think it will.  It's also a completely optional suggestion: I've written code just like this with `nonlocal` etc before (and vaguely wished for a better way to write it), but this other pattern didn't occur to me until just now.  Totally up to you whether you want to leave it as it is (which is fundamentally fine) or make further changes.  You might not even agree with me that my notional change is any better!